apps/webservice/src/app/[workspaceSlug]/_components/CreateDeployment.tsx

```diff
@@ -51,22 +69,39 @@ export const CreateDeploymentDialog: React.FC<{
 }> = ({ children, defaultSystemId = "" }) => {
   const { workspaceSlug } = useParams<{ workspaceSlug: string }>();
   const workspace = api.workspace.bySlug.useQuery(workspaceSlug);
-  const create = api.deployment.create.useMutation();
+  const router = useRouter();
+  const [open, setOpen] = useState(false);
+
+  const create = api.deployment.create.useMutation({
+    onSuccess: (deployment) => {
+      router.refresh();
+      const slug = systems.data?.items.find(
+        (system) => system.id === deployment.systemId,
+      )?.slug;
+      if (slug == null) return;
+      router.push(
+        `/${workspaceSlug}/systems/${slug}/deployments/${deployment.slug}`,
+      );
+      setOpen(false);
+    },
+  });
+
   const form = useForm<DeploymentFormValues>({
+    resolver: zodResolver(deploymentForm),
     defaultValues: {
       systemId: defaultSystemId,
       name: "",
       slug: "",
       description: "",
     },
+    mode: "onChange",
```

Comment:
what happens if we don't set this?

Reply:
It just determines when the validation happens. 

You can have onChange or onSubmit.

Reply:
This makes it validate for each keystroke.

Reply:
We should only validate on submit as it might be annoying to show the error message when they are not finishing typing